Movecraft/src/main/java/net/countercraft/movecraft/listener/CraftAssembleListener.java

```diff
@@ -0,0 +1,75 @@
+package net.countercraft.movecraft.listener;
+
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.SubCraft;
+import net.countercraft.movecraft.events.CraftPilotEvent;
+import net.countercraft.movecraft.events.CraftReleaseEvent;
+import net.countercraft.movecraft.util.MathUtils;
+import org.bukkit.block.Block;
+import org.bukkit.block.TileState;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.persistence.PersistentDataType;
+import org.jetbrains.annotations.NotNull;
+
+public class CraftAssembleListener implements Listener {
```

Comment:
The naming convention should match the event, `CraftPilotListener`.

Reply:
For the moment i limited the stuff in the listener to signs, the mark and unmark methods accept the TileState though. That is, if some addon wants to mark, let's say dispensers (i need to do that for my director changes), they can just do that with the given methods.